packages/typespec-ts/src/utils/clientUtils.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  SdkClient,
+  SdkContext,
+  listClients
+} from "@azure-tools/typespec-client-generator-core";
+import { Namespace } from "@typespec/compiler";
+
+export function getRLCClients(dpgContext: SdkContext): SdkClient[] {
+  const clientsInSdk = listClients(dpgContext);
+  const sdkClientsNamespacesMap = new Map<Namespace, SdkClient[]>();
+  const validClients: SdkClient[] = [];
+  clientsInSdk.forEach((client) => {
+    if (!sdkClientsNamespacesMap.has(client.service)) {
+      sdkClientsNamespacesMap.set(client.service, []);
+    }
+    const clients = sdkClientsNamespacesMap.get(client.service) ?? [];
+    clients.push(client);
+    sdkClientsNamespacesMap.set(client.service, clients);
+  });
+  sdkClientsNamespacesMap.forEach((value, service) => {
+    if (value.length === 1 && value[0]) {
+      // to respect the boundaries that is defined by one @client
+      validClients.push(value[0]);
+    } else {
+      // otherwise use service as the client
+      validClients.push({
+        kind: "SdkClient",
+        name: service.name + "Client",
+        service: service,
+        type: service
+      });
+    }
+  });
```

Comment:
@xirzec @mpodwysocki  I just realize the previous implementation of getRLCClients won't work if we have two @service  endpoints and each endpoint has only one @client, In this case, I think we should also respect the boundaries defined by the @client. Let me know how you think ?

Reply:
If I am understanding correctly, this makes sense to me. 

So, for each service namespace, we respect the client if there is only one or we manufacture a "serviceNameClient" if there are mutliple. Is that correct?

My only question is what happens if initially there is only a single client, and then later on a second one is added? With our current design we would have a breaking change when the single client name is replaced with a service client name, I think.

Reply:
we have provide a `title` configuration in the tspconfig.yaml to customiza the client name. see the example in health insight https://github.com/Azure/autorest.typescript/blob/main/packages/typespec-test/test/healthinsight/tspconfig.yaml#L23, if customers don't want a breaking change, they can always use title to change it back to the previous name. And this title has a higher priority than what's defined in the @client decorators.